packages/typespec-go/package.json

```diff
@@ -50,27 +50,33 @@
     "@azure-tools/cadl-ranch-expect": "~0.14.1",
     "@azure-tools/cadl-ranch-specs": "0.36.1",
     "@azure-tools/typespec-autorest": "0.45.0",
+    "@azure-tools/typespec-azure-core": ">=0.45.0 <1.0.0",
     "@azure-tools/typespec-azure-resource-manager": "0.45.0",
+    "@azure-tools/typespec-client-generator-core": "0.45.4",
```

Comment:
It's also a peer dependency, which tells a host that "we need this dependency at version x.y.z".

Since we're "hostless" when building/running tests, we need to drag this (and others) in via dev dependencies.